### PR TITLE
Use more descriptive accessible headings for search result pages

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes https://github.com/WordPress/openverse/issues/790 by @alexstine (by way of @zackkrida extracting the issue from a tracking issue of several required accessibility fixes)

## Description

**The large number of added lines come from snapshot tests, not actual code changes. See the testing instructions section for details about these.**

The PR adds new sr-only text to the page, as suggested by Alex in the issue. I've chosen to use different variations based on the search type and count of media. In my first attempt, I used two separate h1s. However, this caused an issue with introducing unintended height. Rather than adding additional classes to the second one to prevent that, I decided to just put them inside the same h1, with each bit wrapped in a `span` so I could mark them as aria-hidden or `sr-only` as needed. This appears to work in all my testing (ORCA on Linux, NVDA on Windows and VoiceOver on macOS, the latter two in VMs), but please try to break this to make absolutely sure I've got it right.

- For "all-media" search types, the string is "All results for {query}, {imageResults}, {audioResults}" where the image and audio results are either "No images/audio", "{count} images/audio", or "Over 10000 images/audio". This means the title will inform that the query was for all media types, and the count of results for each media type. That might be too tedious, compared to a unified count, I'm not sure.
- For individual media types, the string is either "No image results for {query}", "1 image result for {query}", "{count} image results for {query}", or "Over 10000 image results for {query}" where "image" can be either image or audio depending on the search type.

## Testing Instructions

Run the application locally using `just p frontend dev` from the root directory of the repository.

Execute a search from the homepage, and using a screen reader, observe the h1 of the page. Confirm it meets the requirements laid out in the issue.

I used snapshot tests rather than individual assertions for each possibility of the accessible text because otherwise I would have to duplicate the translation, which isn't the point. Ideally this would use a diff snapshot test, render a couple "baseline" variations, and then snapshot diff against those to confirm they are all different _in code_, but you can do this by reading the snapshots themselves. This is a small component, so the snapshots are not too cumbersome.

I wanted to use inline snapshots, because I've found those easier to work with in the past. However:
- You cannot use them in an `each` test
- Our trailing whitespace pre-commit fix kept trimming whitespace present in the one that isn't in the `each` block, and I was worried about how many different formatting issue's I'd need to fight against overall. We should disable that trailing whitespace check for anything linted by ruff, prettier, or eslint, though, as it's just duplicating those tools _and_ is less configurable in cases like this... plus, in this case, it was modifying a string literal in the code, which is no good!

Anyway, that's out of scope for this issue. I just wanted to explain the decisions I made there.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/locales/scripts/en.json5
Comment (dhruvkb) [inline frontend/src/locales/scripts/en.json5:467]: Subjective opinion: Number with 'audio' felt wrong.

```suggestion
          count: "{localeCount} audio tracks",
          countMore: "over {localeCount} audio tracks",
```

File: frontend/src/locales/scripts/en.json5
Comment (dhruvkb) [inline frontend/src/locales/scripts/en.json5:440]: IMO, `query` should be surrounded by quotes. I don't know if the screen readers will announce quoted text differently but if they do, quotes can help identify that the phrase in quotes is what the user entered and not a part of the sentence.

**Review by dhruvkb — COMMENTED** [review]
Tried it locally and it works well. I have some subjective opinions but not worth blocking the PR over.

But, if all that's changed is a11y of search headings, the changes to the snapshots are unwarranted. Looking into the changes it seems like the changes are due to font size differences, different shades of grey in checkboxes, and difference in waveforms of audio results. Maybe they are flakes and the checks will still pass if they are reverted?

File: frontend/jest.config.js
Comment (obulat) [reply frontend/jest.config.js:35]: Turning prettier in jest completely (as suggested in [this issue's](https://github.com/vuejs/vue-jest/issues/550#issue-1871220532) workaround) could make the tests run slightly faster. We would still prettify all the code in other steps, so it should be okay?

**Comment by obulat** [comment]
@sarayourfriend, I'm drafting this PR to prevent pings while the comments on prettier and snapshot changes are addressed.

File: frontend/jest.config.js
Comment (sarayourfriend) [reply frontend/jest.config.js:35]: That would be nicer, yes, especially because it's easier to document. Good find, Olga.

File: .pre-commit-config.yaml
Comment (sarayourfriend) [reply .pre-commit-config.yaml:1]: It's the jest snapshot file. It needs to be committed for jest to compare between test runs, but shouldn't be modified with formatting (especially not trailing whitespace) as any minor differences in the strings will cause the snapshot comparison to fail.

File: frontend/src/components/VSearchResultsTitle.vue
Comment (sarayourfriend) [reply frontend/src/components/VSearchResultsTitle.vue:2]: Yes! This is left over from when I used two separate `h1`. It can definitely be removed now.

File: frontend/src/locales/scripts/en.json5
Comment (sarayourfriend) [reply frontend/src/locales/scripts/en.json5:440]: Done in https://github.com/WordPress/openverse/pull/3941/commits/db46a043bcfe4281b2f9329f30436c0b82c74cd3

I tested and can confirm SRs do announce the quotation marks.

File: frontend/src/components/VSearchResultsTitle.vue
Comment (sarayourfriend) [reply frontend/src/components/VSearchResultsTitle.vue]: Done in https://github.com/WordPress/openverse/pull/3941/commits/0929e600b46ce250fca877ff0719970084b02e6b

File: frontend/jest.config.js
Comment (sarayourfriend) [reply frontend/jest.config.js]: Done in https://github.com/WordPress/openverse/pull/3941/commits/c2ccb01d87387e9b4bab42b41ab86b5d1cd37972

File: frontend/src/locales/scripts/en.json5
Comment (sarayourfriend) [reply frontend/src/locales/scripts/en.json5:467]: Done in https://github.com/WordPress/openverse/pull/3941/commits/db46a043bcfe4281b2f9329f30436c0b82c74cd3

**Comment by sarayourfriend** [comment]
@obulat @dhruvkb This is ready for review.

File: frontend/test/unit/specs/components/v-search-results-title.spec.js
Comment (zackkrida) [inline frontend/test/unit/specs/components/v-search-results-title.spec.js:32]: I think there's an opportunity to make this scenario generation a little more readable, at the cost of a more imperative style Here's the general idea and a diff:

```ts
function getScenarios() {
  const counts = [0, 10, 4300, 10000, 10001]
  let scenarios = []
  counts.forEach((count) => {
    counts.forEach((innerCount) => {
      scenarios.push({
        searchType: "all",
        resultCounts: [
          ["image", count],
          ["audio", innerCount],
        ],
      })
    })
    scenarios.push({
      searchType: "image",
      resultCounts: [
        ["image", count],
        ["audio", 0],
      ],
    })
    scenarios.push({
      searchType: "audio",
      resultCounts: [
        ["audio", count],
        ["image", 0],
      ],
    })
  })
  return scenarios.map((scenario) => [JSON.stringify(scenario), scenario])
}
```

And the full diff:

```diff
diff --git a/frontend/test/unit/specs/components/v-search-results-title.spec.js b/frontend/test/unit/specs/components/v-search-results-title.spec.js
index 03c68ce2d..8c4999e56 100644
--- a/frontend/test/unit/specs/components/v-search-results-title.spec.js
+++ b/frontend/test/unit/specs/components/v-search-results-title.spec.js
@@ -12,6 +12,41 @@ const DEFAULT_PROPS = {
   ],
 }
 
+/**
+ * Build test scenarios out of all possible
+ * result count combinations.
+ */
+function getScenarios() {
+  const counts = [0, 10, 4300, 10000, 10001]
+  let scenarios = []
+  counts.forEach((count) => {
+    counts.forEach((innerCount) => {
+      scenarios.push({
+        searchType: "all",
+        resultCounts: [
+          ["image", count],
+          ["audio", innerCount],
+        ],
+      })
+    })
+    scenarios.push({
+      searchType: "image",
+      resultCounts: [
+        ["image", count],
+        ["audio", 0],
+      ],
+    })
+    scenarios.push({
+      searchType: "audio",
+      resultCounts: [
+        ["audio", count],
+        ["image", 0],
+      ],
+    })
+  })
+  return scenarios.map((scenario) => [JSON.stringify(scenario), scenario])
+}
+
 describe("VSearchResultsTitle", () => {
   const getOptions = ({ props, ...options } = {}) => ({
     ...options,
@@ -27,37 +62,7 @@ describe("VSearchResultsTitle", () => {
   })
 
   describe("accessible heading", () => {
-    const cs = [0, 10, 4300, 10000, 10001]
-
-    const scenarios = [
-      ...cs.reduce(
-        (all, i) => [
-          ...all,
-          ...cs.map((a) => ({
-            searchType: "all",
-            resultCounts: [
-              ["image", i],
-              ["audio", a],
-            ],
-          })),
-        ],
-        []
-      ),
-      ...cs.map((count) => ({
-        searchType: "image",
-        resultCounts: [
-          ["image", count],
-          ["audio", 0],
-        ],
-      })),
-      ...cs.map((count) => ({
-        searchType: "audio",
-        resultCounts: [
-          ["audio", count],
-          ["image", 0],
-        ],
-      })),
-    ].map((scenario) => [JSON.stringify(scenario), scenario])
+    let scenarios = getScenarios()
 
     it.each(scenarios)("%s", (_, { searchType, resultCounts }) => {
       const { container } = render(

```

File: frontend/src/locales/scripts/en.json5
Comment (zackkrida) [reply frontend/src/locales/scripts/en.json5:627]: @sarayourfriend I believe "tracks" is still missing in the cases where the string is `All results for "zack", 10 images and 10 audio`.

**Review by zackkrida — CHANGES_REQUESTED** [review]
Looks great. Requesting one small change for the last place where "tracks" is missing after "audio", and a non-blocking suggestion for code readability in the unit test.

**Review by zackkrida — APPROVED** [review]
Great :+1: I appreciate the unit test improvements as well, like adding the translation job and excluding the tailwind prettier plugin

**Review by obulat — APPROVED** [review]
I tested this PR on NVDA and VoiceOver.

NVDA reads the heading, and you can get to the heading using the `H` shortcut.
In VoiceOver, you have to navigate inside the heading element to read the heading using the shortcut (VO+command+H). If you try the same shortcut when the VO focus area is in the content area (one level above), it says "Heading not found". I'm not sure if it's supposed to be like that, or if I'm using VO incorrectly, or if VO has a bug, but I couldn't find a way to navigate through the headings on the search page.

**Comment by zackkrida** [comment]
A few snapshots need updating after the merge of #3967

**Comment by sarayourfriend** [comment]
I had to squash this PR, the back-and-forth commits were making rebases a pain. I should be able to merge this now so long as CI passes, there are no changes other than updates to the snapshots that Zack mentioned.

**Pull request merged**
